src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -372,7 +372,113 @@ protected void InitializeReferences()
             RegexRunner runner = Interlocked.Exchange(ref _runner, null) ?? CreateRunner();
             try
             {
-                return runner.Scan(this, input, beginning, beginning + length, startat, prevlen, quick, internalMatchTimeout);
+                runner.InitializeTimeout(internalMatchTimeout);
+                runner.runtext = input;
+                ReadOnlySpan<char> span = input.AsSpan(beginning, length);
+                runner.InitializeForScan(this, span, startat - beginning, quick);
+
+                int stoppos = RightToLeft ? 0 : span.Length;
+
+                // If previous match was empty or failed, advance by one before matching.
+                if (prevlen == 0)
+                {
+                    if (runner.runtextstart == stoppos)
+                    {
+                        return RegularExpressions.Match.Empty;
+                    }
+
+                    runner.runtextpos += RightToLeft ? -1 : 1;
+                }
```

Comment:
We will eventually, but we don't for `IsMatch`. What I did in this PR is basically the minimum to support isMatch overload which basically is: return null if you found a match, and Match.Empty otherwise. I plan to add the rest of the body of this method when I add Count, Enumerate, etc.